### PR TITLE
[docs]: add the Edit in Gitpod button.

### LINKED ISSUE #380: [Edit|Improve] this page links on the documentation
If there was a spelling error then the community or Gitpoder should be able to "edit (propose changes via PR) this file on GitHub" from the page they are viewing.

### PR DESCRIPTION
**Pull request by nisarhassan12**
Fixes #380 

This is how it looks:

image

<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/500"></a>

### CODE REVIEW

**Comment by nisarhassan12** [comment]
@ChristinFrohne at the moment it has no hover like the Open in GitPod button used on the READMEs on GitHub but since it lives on the site we can instead of using and SVG re-create the button ourselves and the add hover/focus effect.

**Comment by ChristinFrohne** [comment]
@nisarhassan12 yes, please use the specs that MAD provided (see Figma file)
Bildschirmfoto 2021-05-27 um 15 12 36

**Comment by nisarhassan12** [comment]
@ChristinFrohne I have made the requested changes please have a look again. Thanks 

Regarding the opposition of colour b/w hover and no hover was decided earlier in April because the colour contrast was not enough same is true for the browser extension buttons:

image

**Comment by ChristinFrohne** [comment]
Great, could you use the small button style that MAD provided @nisarhassan12 ? Currently, it takes a bit too much attention. Thanks!

**Comment by nisarhassan12** [comment]
@ChristinFrohne I have made the button small as per mad guidelines. 

image

File: src/components/docs/edit-in-gitpod.svelte [hunk 1]
```diff
@@ -1,14 +1,19 @@
 <script lang="ts">
+  import { page } from "$app/stores";
   import LogoTextless from "../svgs/logo-textless.svelte";
-  export let pagePath;
```
Comment (mikenikles) [inline src/components/docs/edit-in-gitpod.svelte]: In my opinion, fewer props are better as they make components more reusable. If we expect this prop, the user of the component needs to know what the value of `pagePath` needs to be and how to fetch it.

In this particular case, we can easily get that value from within the component.

File: src/components/docs/edit-in-gitpod.svelte [hunk 2]
```diff
@@ -1,14 +1,19 @@
...
     "https://gitpod.io#https://github.com/gitpod-io/website/tree/main/src/routes";
-  const currentPage = pagePath === "/docs" ? `${pagePath}/index` : pagePath;
+  const currentPage =
+    $page.path === "/docs" ? `${$page.path}/index` : $page.path;
   const href = `${BASE_PATH}${currentPage}.md`;
 </script>
 
 <style>
+  a::after {
+    display: none !important;
```
Comment (mikenikles) [inline src/components/docs/edit-in-gitpod.svelte:14]: I needed to this overwrite that new external link arrow. See below for an explanation 😄.

File: src/routes/docs/$layout.svelte [hunk 3]
```diff
@@ -202,7 +201,6 @@
   <div class="doc-contents">
     <Search />
     <MobileMenu {MENU} />
-    <EditInGitpod pagePath={$page.path} />
```
Comment (mikenikles) [inline src/routes/docs/$layout.svelte]: This didn't work on the deployed, static site. The Gitpod URL always pointed to `/docs/index.md`. I moved the button to the `src/components/docs/edit-in-gitpod.svelte` file, which is the layout for each individual page, hence the path changes dynamically when each page is generated for the static site.

File: src/components/docs/edit-in-gitpod.svelte (hunk 1 above)
Comment (nisarhassan12) [reply src/components/docs/edit-in-gitpod.svelte]: Thanks. Got it.

**Pull request merged**